src/psycopmlutils/timeseriesflattener/data_integrity.py

```diff
@@ -0,0 +1,284 @@
+    # create subfolder for outcome specific checks
+    outcome_checks_dir = out_dir / "outcomes"
+    if not outcome_checks_dir.exists():
+        outcome_checks_dir.mkdir()
+
+    ###################
+    #### DATA INTEGRITY
+    ###################
+
+    if check_data_integrity:
+        msg.info("Running data integrity checks...")
+        # Only running data integrity checks on the training set to reduce the
+        # chance of any form of peaking at the test set
+        train_predictors, train_outcomes = load_split_predictors_and_outcomes(
+            path=path,
+            split="train",
+            include_id=False,
+            nrows=nrows,
+        )
+        ds = Dataset(df=train_predictors, datetime_name="timestamp")
+
+        # Running checks that do not require a label
+        integ_suite = data_integrity()
+        suite_results = integ_suite.run(ds)
+        suite_results.save_as_html(str(out_dir / "data_integrity.html"))
+
+        # Running checks that require a label for each outcome
+        label_checks = label_integrity_checks()
+        for outcome_column in train_outcomes.columns:
+            msg.info(f"Running data integrity for {outcome_column}")
+            ds = Dataset(
+                df=train_predictors,
+                datetime_name="timestamp",
+                label=train_outcomes[outcome_column],
+            )
+            suite_results = label_checks.run(ds)
+            suite_results.save_as_html(
+                str(outcome_checks_dir / f"{outcome_column}_check.html"),
+            )
+
+        msg.good("Finshed data integrity checks!")
+
+    #####################
+    #### SPLIT VALIDATION
+    #####################
+    if check_split_integrity:
+        msg.info("Running split validation...")
+        # Running data validation checks on train/val and train/test splits that do not
+        # require a label
+        validation_suite = custom_train_test_validation()
+
+        train_predictors, train_outcomes = load_split_predictors_and_outcomes(
+            path=path,
+            split="train",
+            include_id=True,
+            nrows=nrows,
+        )
+        val_predictors, val_outcomes = load_split_predictors_and_outcomes(
+            path=path,
+            split="val",
+            include_id=True,
+            nrows=nrows,
+        )
+        test_predictors, test_outcomes = load_split_predictors_and_outcomes(
+            path=path,
+            split="test",
+            include_id=True,
+            nrows=nrows,
+        )
+
+        train_ds = Dataset(
+            train_predictors,
+            index_name="dw_ek_borger",
+            datetime_name="timestamp",
+        )
+        val_ds = Dataset(
+            val_predictors,
+            index_name="dw_ek_borger",
+            datetime_name="timestamp",
+        )
+        test_ds = Dataset(
+            test_predictors,
+            index_name="dw_ek_borger",
+            datetime_name="timestamp",
+        )
```

Comment:
I wonder whether this should be refactored out to a dict with a structure like:

```
data = {"train": {"outcomes": [...], "predictors": [...],}}
```

and so forth, nested. Then we could load straight into the dict. 

It's a bit hard because `load_split_predictors_and_outcomes` returns a tuple, but maybe it should be refactored into two different functions? 

Then it better fits into a set of utilities, and we could use it in the training script as well! :-)